scraper/PageScraper.ts

```diff
@@ -0,0 +1,340 @@
+import * as puppeteer from 'puppeteer'
+import {
+  TimetableUrl,
+  Chunk,
+  PageData,
+  Term,
+  Course,
+  Warning,
+  ClassWarnings,
+  ClassesByTerm,
+  TimetableData,
+  CourseInfo,
+  TermFinderReference,
+} from './interfaces'
+import { formatDates, keysOf } from './helper'
+import { parseCourseInfoChunk, getCourseHeadData } from './ChunkScraper'
+import { parseClassChunk, classTermFinder } from './ClassScraper'
+
+interface getDataUrlsParams {
+  page: puppeteer.Page
+  regex: RegExp
+}
+
+/**
+ * Gets all the urls in the data class on page: page,
+ * given regex: regex.
```

Comment:
Don't need to repeat yourself

```suggestion
 * given regex.
```

Reply:
Same for the line above

Reply:
```suggestion
 * Gets all the urls in the data class on page matching regex
```